src/Microsoft.OData.Core/MultipartMixed/ODataMultipartMixedBatchWriter.cs

```diff
@@ -217,15 +240,24 @@ protected override void WriteStartChangesetImplementation(string changeSetId)
             this.WritePendingMessageData(true);
 
             // create the new request operation
+            // For Multipart batch format, validate dependsOnIds if it is user explicit input, otherwise skip validation
+            // when it is implicitly derived per protocol.
             ODataBatchOperationRequestMessage operationRequestMessage = BuildOperationRequestMessage(
                 this.RawOutputContext.OutputStream,
-                method, uri, contentId, /*groupId*/null, dependsOnIds, ODataFormat.Batch);

```

Comment:
>/*groupId*/null [](start = 40, length = 15)

Why did we start passing null, rather than ODataMultipartMixedBatchWriterUtils.GetChangeSetIdFromBoundary(this.changeSetBoundary)?

Reply:
groupId is introduced with JSON Batch. For MultipartMixed batch, we have changeset boundary instead and we originally considered groupId value is always null. And then we decided to use the value extracted from changeset boundary as group Id, likely from one of your comments.

---
In reply to: [160844104](https://github.com/OData/odata.net/pull/1020#discussion_r160844104) [](ancestors = 160844104)

Reply:
Sorry; I was reacting to an apparent change to pass null here, but that was an old iteration.  Now we pass GetChangeSetIdFromBoundary, which is what I would expect.

---
In reply to: [160846771](https://github.com/OData/odata.net/pull/1020#discussion_r160846771) [](ancestors = 160846771,160844104)